src/System.Slices/src/System/Span.cs

```diff
@@ -283,45 +288,25 @@ public bool ReferenceEquals(Span<T> other)
 
         public override int GetHashCode()
         {
-            // TODO: Write something better
-            uint hash = unchecked((uint)Length);
-            foreach (T el in this)
+            var hashCode = Length ^ Offset.GetHashCode();
```

Comment:
At a minimum this seems like a not great hashing function because we don't mix in the native pointer for spans over un-managed data, so all spans backed by native data with the same offset and length will have the same hashcode.

Maybe start from either the raw pointer (either truncated down to an int (this is what IntPtr.GetHashCode does) or XOR the high and low 32 bits together) or Object.GetHashCode() and then mix in the length and offset via multiplying the value by a prime (e.g. 33) and the adding in each piece of data.  This would be similar to how `string.GetHashCode()` works.